yourcode/udp-sender.cc

```diff
@@ -33,10 +33,14 @@ int main( int argc, char *argv[] )
   /* XXX your code here */
   
   /* construct UDP socket */
+  UDPSocket sock;
 
   /* connect the socket to the given host and port */
+  uint16_t i = stoi(port, nullptr);
```

Comment:
Nice! What happens if you just pass the `port` string directly to the Address constructor in the `sock.connect` line? (I.e. using the string/string version of the Address constructor?) Does it still work?